docs/core/diagnostics/metrics-strongly-typed.md

```diff
@@ -0,0 +1,189 @@
+
+All three metric definitions `CreateLatency`, `CreateTotalCount`, and `CreateTotalFailures` use `MetricTags` as their tag object type. This means the generated metric types (`Latency`, `TotalCount`, and `TotalFailures`) will all expect a `MetricTags` instance when recording data. **Each of these metrics will have the same set of tag names:** `Dim1DimensionName`, `Operation`, `Dim2`, `Dim3`, and `DimensionNameOfParentOperation`.
+
+The code below shows how to create and use these metrics in a class:
+
+```csharp
+internal class MyClass
+{
+    private readonly Latency _latencyMetric;
+    private readonly TotalCount _totalCountMetric;
+    private readonly TotalFailures _totalFailuresMetric;
+
+    public MyClass(Meter meter)
+    {
+        // Create metric instances using the source-generated factory methods
+        _latencyMetric = Metric.CreateLatency(meter);
+        _totalCountMetric = Metric.CreateTotalCount(meter);
+        _totalFailuresMetric = Metric.CreateTotalFailures(meter);
+    }
+
+    public void DoWork()
+    {
+        var stopwatch = new Stopwatch();
+        stopwatch.Start();
+        bool requestSuccessful = true;
+        // ... perform some operation ...
+        stopwatch.Stop();
+
+        // Create a tag object with values for all tags
+        var tags = new MetricTags
+        {
+            Dim1 = "Dim1Value",
+            Operation = Operations.Operation1,
+            ParentOperationName = "ParentOpValue",
+            ChildTagsObject = new MetricChildTags
+            {
+                Dim2 = "Dim2Value",
+            },
+            ChildTagsStruct = new MetricTagsStruct
+            {
+                Dim3 = "Dim3Value"
+            }
+        };
+
+        // Record the metric values with the associated tags
+        _latencyMetric.Record(stopwatch.ElapsedMilliseconds, tags);
+        _totalCountMetric.Add(1, tags);
+        if (!requestSuccessful)
+        {
+            _totalFailuresMetric.Add(1, tags);
+        }
+    }
+}
+```
+
+In the `MyClass.DoWork` method above, a `MetricTags` object is populated with values for each tag. This single `tags` object is then passed to all three instruments when recording data. The `Latency` metric (a histogram) records the elapsed time, and both counters (`TotalCount` and `TotalFailures`) record occurrence counts. Because all metrics share the same tag object type, the tags (`Dim1DimensionName`, `Operation`, `Dim2`, `Dim3`, `DimensionNameOfParentOperation`) are present on every measurement.
+
+## Performance considerations
+
+Using strongly-typed tags via source generation adds no overhead compared to using metrics directly. If you need to further minimize allocations for very high-frequency metrics, consider defining your tag object as a `struct` (value type) instead of a class. Using a struct for the tag object can avoid heap allocations when recording metrics, since the tags would be passed by value.
```

Comment:
```suggestion
Using strongly-typed tags via source generation adds no overhead compared to using metrics directly. If you need to further minimize allocations for very high-frequency metrics, consider defining your tag object as a `struct` (value type) instead of a `class`. Using a `struct` for the tag object can avoid heap allocations when recording metrics, since the tags would be passed by value.
```